chatterbot/utils/parsing.py

```diff
@@ -0,0 +1,562 @@
+import re
+from datetime import timedelta, date, datetime
+import calendar
+
+# Variations of dates that the parser can capture
+year_variations = ['year', 'years', 'yrs']
+day_variations = ['days', 'day']
+minute_variations = ['minute', 'minutes', 'mins']
+week_variations = ['weeks', 'week', 'wks']
+
+# Variables used for RegEx Matching
+day_names = 'monday|tuesday|wednesday|thursday|friday|saturday|sunday'
+month_names = 'january|february|march|april|may|june|july|august|september|october|november|december|jan|feb|mar|apr|may|jun|jul|aug|sep|oct|nov|dec'
+day_nearest_names = 'today|yesterday|tomorrow|tonight|tonite'
+numbers = "(^a(?=\s)|one|two|three|four|five|six|seven|eight|nine|ten| \
+                    eleven|twelve|thirteen|fourteen|fifteen|sixteen|seventeen| \
+                    eighteen|nineteen|twenty|thirty|forty|fifty|sixty|seventy|eighty| \
+                    ninety|hundred|thousand)"
+re_dmy = '(' + "|".join(day_variations + minute_variations + year_variations + week_variations) + ')'
+re_duration = '(before|after|earlier|later|ago|from\snow)'
+re_year = "(?<=\s)\d{4}|^\d{4}"
+re_timeframe = 'this|next|following|previous|last'
+re_ordinal = 'st|nd|rd|th|first|second|third|fourth|fourth|' + re_timeframe
+re_time = '(?P<hour>\d+)(\:(?P<minute>\d{2}))?(?P<convention>am|pm|a\.m\.|p\.m\.)?'
+re_separator = 'of|at|on'
+
+# A list tuple of regular expressions / parser fn to match
+# The order of the match in this list matters, So always start with the widest match and narrow it down
+regex = [
+    (re.compile(
+        r'''
+        (
+            ((?P<dow>%s)[,\s]\s*)? #Matches Monday, 12 Jan 2012, 12 Jan 2012 etc
+            (?P<day>\d{1,2}) # Matches a digit
+            (%s)?
+            [-\s] # One or more space
+            (?P<month>%s) # Matches any month name
+            [-\s] # Space
+            (?P<year>\d{4}) # Year
+            ((\s|,\s|\s(%s))?\s*(%s))?
+        )
+        '''% (day_names, re_ordinal, month_names, re_separator, re_time),
+        (re.VERBOSE | re.IGNORECASE)
+        ),
+        lambda (m, base_date): datetime(
```

Comment:
Sure. I am going to add support for other languages and will push the changes soon.